addons/mass_mailing_sms/static/src/js/fields_sms_widget.js

```diff
@@ -0,0 +1,92 @@
+/** @odoo-module **/
+
+import { _t } from "web.core";
+import SmsWidget from '@sms/js/fields_sms_widget';
+
+const TEXT_URL_REGEX = /https?:\/\/[\w@:%.+&~#=/-]+(?:\?\S+)?/g;  // from tools.mail.TEXT_URL_REGEX
+
+/**
+ * Override to provide extra characters count information to
+ * consider links converted with link_tracker and opt-out
+ * link if the option is selected.
+ */
+SmsWidget.include({
+    events: _.extend({}, SmsWidget.prototype.events || {}, {
+        'focusin': '_onClick',
+    }),
+    /**
+     *@override
+     */
+    willStart: async function () {
+        await this._super.apply(this, arguments);
+        if (this.model === "mailing.mailing" && this.mode === 'edit') {
+            const { optOutLink, link } = await this._rpc({
+                model: 'mailing.mailing',
+                method: 'get_chars_counts',
+                args: [this.res_id]
+            });
+            this.charsCountData = { optOutLink, link };
+            this.optOutEnabled = this.recordData.sms_sms_allow_unsubscribe;
+            this.noticeLinksReplaced = false;
+        }
+    },
+    /**
+     * Add the characters for opt-out link if enabled.
+     *
+     * @override
+     */
+    _getNbrChar: function () {
+        let nbrChar = this._super.apply(this, arguments);
+        if (this.charsCountData) {
+            // this is the simplest way to read from another field in a legacy fix.
+            const sms_allow_unsubscribe_el = document.querySelector('div[name="sms_allow_unsubscribe"] input');
+            if (sms_allow_unsubscribe_el) {
+                this.optOutEnabled = sms_allow_unsubscribe_el.checked;
```

Comment:
Does it really belong to '_getNbrChar' method ? Shouldn't it be on its side, more like a state update ?